### PR TITLE
Added brief readme on training example scripts

### PR DESCRIPTION
**Pull request by shanjiaz**
Added training scripts readme.

### CODE REVIEW

File: examples/training/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Training Examples
+
+We provide three example bash scripts that demonstrate the training workflow. Each script downloads one or two small toy dataset of precomputed hidden states (50 samples) and runs `scripts/train.py` to train a model on the datasets.
```
Comment (dsikka) [inline examples/training/README.md]: `train a speculator`

File: examples/training/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+```bash
+bash examples/training/gpt_oss_20b_ultrachat.sh
+bash examples/training/llama3_8b_sharegpt.sh
+bash examples/training/qwen3_8b_sharegpt_ultrachat.sh
+```
+
+## Expected logs
+
+When running, you're see logs containing color-coded information like current epoch, loss, accuracy, conditional accuracy, learning_rate, etc. Example logs
```
Comment (dsikka) [inline examples/training/README.md]: you will see, not you're

File: examples/training/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+[20:17:13] INFO     train/loss_0=1.980, train/full_acc_0=0.029, train/cond_acc_0=0.029, train/loss_1=1.970, train/full_acc_1=9.45e-04, train/cond_acc_1=0.033, train/loss_2=2.015, train/full_acc_2=0.00e+00, train/cond_acc_2=0.00e+00, train/loss=5.965, epoch=0, lr=2.47e-05             trainer.py:191
+```
+
+## Output
```
Comment (dsikka) [inline examples/training/README.md:32]: Once training is complete ...

File: examples/training/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+The scripts create an `./output` directory in the current working directory like the following:
+
+```
+output/
+├── data/         # Downloaded HuggingFace datasets
+└── checkpoints/  # Training checkpoints
+```
+
+The `./data` directory saves the huggingface datasets downloaded and the 'checkpoints' directory saves checkpoints produced by the training scripts. These checkpoints can served in vLLM directly. (Results won't be great since they're only trained on 50 examples)
```
Comment (dsikka) [inline examples/training/README.md]: The trained speculator is saved in the speculator format and can directly be served in vllm using `vllm serve <path_to_speculator>`. Note: this speculator is only trained on 50 samples so accuracy will not be high.

**Review by dsikka — CHANGES_REQUESTED** [review]

File: examples/training/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Training Examples
+
+We provide three example bash scripts that demonstrate the training workflow. Each script downloads one or two small toy datasets of precomputed hidden states (50 samples) and runs `scripts/train.py` to train a speculator model on the datasets.
```
Comment (dsikka) [inline examples/training/README.md]: sorry one more nit, can we list using the EAGLE3 algorithm?

File: examples/training/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+When running, you'll see logs containing color-coded information like current epoch, loss, accuracy, conditional accuracy, learning_rate, etc. Example logs:
+
+```
+[20:17:13] INFO     train/loss_0=1.980, train/full_acc_0=0.029, train/cond_acc_0=0.029, train/loss_1=1.970, train/full_acc_1=9.45e-04, train/cond_acc_1=0.033, train/loss_2=2.015, train/full_acc_2=0.00e+00, train/cond_acc_2=0.00e+00, train/loss=5.965, epoch=0, lr=2.47e-05             trainer.py:191
+```
+
+## Output
+
+Once training is complete, the scripts create an `./output` directory in the current working directory like the following:
+
```
Comment (dsikka) [inline examples/training/README.md:35]: Can we add a generic blurb from here as well: https://github.com/vllm-project/speculators/tree/main/examples/data_generation_and_training#training

File: examples/training/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+output/
+├── data/         # Downloaded HuggingFace datasets
+└── checkpoints/  # Training checkpoints
+```
+
+The `./data` directory saves the huggingface datasets downloaded and the 'checkpoints' directory saves checkpoints produced by the training script. The trained speculator is saved in the speculator format and can directly be served in vLLM using `vllm serve <path_to_speculator>`. Note: this speculator is only trained on 50 samples so accuracy will not be high.
```
Comment (dsikka) [inline examples/training/README.md]: Can we also link to the training script, for more details: https://github.com/vllm-project/speculators/blob/main/scripts/README.md

File: examples/training/README.md (hunk 5 above)
Comment (shanjiaz) [reply examples/training/README.md]: yes!

File: examples/training/README.md (hunk 6 above)
Comment (shanjiaz) [reply examples/training/README.md:35]: yes! Added as the first example.

File: examples/training/README.md (hunk 7 above)
Comment (shanjiaz) [reply examples/training/README.md]: linked it in the training script section. We want the read me on branch [v0.4.0+rhaiis](https://github.com/vllm-project/speculators/tree/v0.4.0%2Brhaiis) right?

**Review by dsikka — APPROVED** [review]

File: examples/training/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Training Examples
+
+We provide three example bash scripts that demonstrate the training workflow. Each script downloads one or two small toy datasets of precomputed hidden states (50 samples) and runs `scripts/train.py` to train a EAGLE3 speculator model on the datasets.
+
+## Training Script
+
+Train an Eagle3 draft model or speculator. Currently, training is supported for:
+
+1. Single-Layer and Multi-Layer Draft Models for Non-MoE models
+2. Single-Layer and Multi-Layer Draft Models of certain Non-Vision MoEs For a full list of models with support, see: https://github.com/vllm-project/speculators/blob/main/README.md
```
Comment (fynnsu) [inline examples/training/README.md:10]: This makes it sound like we support "Non-MoE Vision models" which I don't think we do? Can we just simplify this to Single or Multi layer draft models for text only LLMs? Or something like that?

File: examples/training/README.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+Once training is complete, the scripts create an `./output` directory in the current working directory like the following:
+
+```
+output/
+├── data/         # Downloaded HuggingFace datasets
+└── checkpoints/  # Training checkpoints
+```
+
+The `data` directory saves the huggingface datasets downloaded and the `checkpoints` directory saves checkpoints produced by the training script. The trained speculator is saved in the speculator format and can directly be served in vLLM using `vllm serve <path_to_speculator>`. Note: this speculator is only trained on 50 samples so accuracy will not be high.
```
Comment (fynnsu) [inline examples/training/README.md]:
```suggestion
The `data` directory saves the huggingface datasets downloaded and the `checkpoints` directory saves checkpoints produced by the training script. The trained speculator is saved in the speculator format and can directly be served in vLLM using `vllm serve <path_to_speculator>`. Note: this speculator is only trained on 50 samples so acceptance rates will not be high.
```

Comment (fynnsu) [reply examples/training/README.md]: Maybe we should also suggest they use one of the models we trained from the HF collection instead?

File: examples/training/README.md (hunk 8 above)
Comment (shanjiaz) [reply examples/training/README.md:10]: A little bit, this is what we say in main tho https://github.com/vllm-project/speculators/blob/main/scripts/README.md, not sure if we should change our wording to differ from main.

File: examples/training/README.md (hunk 9 above)
Comment (shanjiaz) [reply examples/training/README.md]: the collection is mentioned in Training Script section. Do you think maybe we should add here as well?

File: examples/training/README.md (hunk 8 above)
Comment (fynnsu) [reply examples/training/README.md:10]: Yeah, I guess I just found this kind of confusing.
